snap/helpers/app.ini

```diff
@@ -0,0 +1,69 @@
+APP_NAME    = Gitea: Go Git Service
+RUN_USER    = root
+RUN_MODE    = prod
+CUSTOM_PATH = SNAP_DIR_DATA/custom
+
+[server]
+DOMAIN            = localhost
+PROTOCOL          = http
+HTTP_PORT         = 3001
+ROOT_URL          = http://localhost:3001/
+DISABLE_SSH       = false
+SSH_PORT          = 22
+OFFLINE_MODE      = true
+STATIC_ROOT_PATH  = SNAP_DIR_DATA/static
+APP_DATA_PATH     = SNAP_DIR_COMMON/data
+SSH_ROOT          = SNAP_DIR_COMMON/ssh
+SSH_KEY_TEST_PATH = SNAP_DIR_DATA/sshkeytest
+CERT_FILE         = SNAP_DIR_DATA/certs/cert.pem
+KEY_FILE          = SNAP_DIR_DATA/certs/key.pem
```

Comment:
CERT_FILE and KEY_FILE aren't need if protocol is http. Maybe we could generate some at install and default to https ?

Reply:
A fair bit of time has gone by since the writing and a lot of new features have come to snapcraft, so I'll re-check this. These entries were added back then because I judged including a full config-writer or something hack-ish would've been more costly than having these two lines in there.

Reply:
I've kept them in, and in a way 'made it worse' by moving to the snap hooks for configuration.

Doing that I'm using the defaults from app.ini.sample, overwriting the parts necessary for a snap to function and adding them all to the app.ini as used by gitea.

I personally prefer this, as it's now possible to configure everything via `sudo snap set category-subcategory.option`, but it has some drawbacks at the moment, as I didn't want to touch the config system:

* Cateogories in the style of `category.subcategory` are now `category-subcategory`
* Options containing an underscore have a '99' in its place (`ssh.minimum_key_size` is `ssh-minimum99key99size` for snapctl)
* New defaults do not apply automatically
* Service-restart takes longer (it re-parses the whole app.ini every time)

All options can still be changed manually in the app.ini file and will be added to `snapctl` upon service-restart.

Not sure which way is preferred, but unless we'll touch the config system for this, there's not much that can be done (aside from completely leaving it alone and up to the user to set everything).

I would love to have it fully integrated into the snap config system as an option, but that would require a lot of work:

* Category names are only allowed to be lower case and can only contain dashes
* There is no programmatic interface from what I can see, all has to be done with `exec()` calls.

ini files are regrettably hard to convert to the json-based snapctl config format.